qiskit/aqua/operators/primitive_ops/summed_pauli_op.py

```diff
@@ -0,0 +1,384 @@
+        # TODO: optimize for some special cases
+        from ..evolutions.evolved_op import EvolvedOp
+
+        return EvolvedOp(self)
+
+    def to_instruction(self) -> Instruction:
+        return self.to_matrix_op().to_circuit().to_instruction()  # type: ignore
+
+    def to_pauli_op(self, massive: bool = False) -> OperatorBase:
+        from .pauli_op import PauliOp
+
+        def to_real(x):
+            return x.real if np.isreal(x) else x
+
+        def to_native(x):
+            return x.item() if isinstance(x, np.generic) else x
+
+        if len(self.primitive) == 1:
+            return PauliOp(
+                Pauli(x=self.primitive.table.X[0], z=self.primitive.table.Z[0]),  # type: ignore
+                to_native(to_real(self.primitive.coeffs[0])) * self.coeff,  # type: ignore
+            )
+        return SummedOp(
+            [
+                PauliOp(
+                    Pauli(x=s.table.X[0], z=s.table.Z[0]),
+                    to_native(to_real(s.coeffs[0])),
+                )
+                for s in self.primitive
+            ],
+            coeff=self.coeff,
+        )
+
+    def __getitem__(self, offset: Union[int, slice]) -> "SummedPauliOp":
+        """Allows array-indexing style access to the ``SummedPauliOp``.
+
+        Args:
+            offset: The index of ``SummedPauliOp``.
+
+        Returns:
+            The ``SummedPauliOp`` at index ``offset``,
+        """
+        return SummedPauliOp(self.primitive[offset], self.coeff)
+
+    def __len__(self) -> int:
+        """Length of ``SparsePauliOp``.
+
+        Returns:
+            An int equal to the length of SparsePauliOp.
+        """
+        return len(self.primitive)
+
+    # pylint: disable=arguments-differ
+    def reduce(
+            self, atol: Optional[float] = None, rtol: Optional[float] = None
+    ) -> "SummedPauliOp":
+        """Simplify the primitive ``SparsePauliOp``.
+
+        Args:
+            atol: Absolute tolerance for checking if coefficients are zero (Default: 1e-8).
+            rtol: Relative tolerance for checking if coefficients are zero (Default: 1e-5).
+
+        Returns:
+            The simplified ``SummedPauliOp``.
+        """
+        return SummedPauliOp(
+            self.primitive.simplify(atol=atol, rtol=rtol), self.coeff  # type: ignore
+        )
+
+    def to_spmatrix(self) -> spmatrix:
+        """Returns SciPy sparse matrix representation of the ``SummedPauliOp``.
+
+        Returns:
+            CSR sparse matrix representation of the ``SummedPauliOp``.
+
+        Raises:
+            ValueError: invalid parameters.
+        """
+        return self.primitive.to_matrix(sparse=True) * self.coeff  # type: ignore
+
+    @property
+    def oplist(self) -> List[OperatorBase]:
+        """The list of operators.
+        This function is for compatibility with ``SummedOp``.
+
+        Return:
+            The list of operators.
+        """
+        return self.to_pauli_op().oplist  # type: ignore
```

Comment:
Having seen your comment on 'oplist' for SummedOp compatibility, and also the comments around empty and len=1,  it seems that to_pauli_op().oplist will not return a List as it states it does in that case since the to_pauli_op returns a single PauliOp so this method, I think should check and make a list from the single PauliOp if that is the case.

Reply:
hmm, you are right. I think this type of conversion is dangerous.
I want to remove oplist from SummedPauliOp. And I will consider rename `SummedPauliOp`